pkg/validations/createcluster/createcluster_test.go

```diff
@@ -0,0 +1,86 @@
+package createcluster_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/aws/eks-anywhere/internal/test"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/gitops/flux"
+	providermocks "github.com/aws/eks-anywhere/pkg/providers/mocks"
+	"github.com/aws/eks-anywhere/pkg/validations/createcluster"
+	"github.com/aws/eks-anywhere/pkg/validations/mocks"
+	validatormocks "github.com/aws/eks-anywhere/pkg/workflows/interfaces/mocks"
+)
+
+type createClusterValidationTest struct {
+	clusterConfigFile string
+	clusterSpec       *cluster.Spec
+	ctx               context.Context
+	provider          *providermocks.MockProvider
+	flux              *flux.Flux
+	docker            *mocks.MockDockerExecutable
+	validator         *validatormocks.MockValidator
+}
+
+func newValidateTest(t *testing.T) *createClusterValidationTest {
+	mockCtrl := gomock.NewController(t)
+	clusterConfigFile := "testdata/cluster_vsphere.yaml"
+	provider := providermocks.NewMockProvider(mockCtrl)
+	docker := mocks.NewMockDockerExecutable(mockCtrl)
+	validator := validatormocks.NewMockValidator(mockCtrl)
+	return &createClusterValidationTest{
+		clusterConfigFile: clusterConfigFile,
+		clusterSpec:       test.NewFullClusterSpec(t, clusterConfigFile),
+		ctx:               context.Background(),
+		provider:          provider,
+		docker:            docker,
+		validator:         validator,
+	}
+}
+
+func (c *createClusterValidationTest) parseClusterSpecFromFile(t *testing.T, clusterConfigFile string) {
+	c.clusterSpec = test.NewFullClusterSpec(t, clusterConfigFile)
+}
+
+func (c *createClusterValidationTest) expectValidProvider() {
+	c.provider.EXPECT().SetupAndValidateCreateCluster(c.ctx, c.clusterSpec).Return(nil).AnyTimes()
+	c.provider.EXPECT().Name().Return("vsphere").AnyTimes()
+}
+
+func (c *createClusterValidationTest) expectValidDockerExec() {
+	c.docker.EXPECT().Version(c.ctx).Return(21, nil).AnyTimes()
+	c.docker.EXPECT().AllocatedMemory(c.ctx).Return(uint64(6200000001), nil).AnyTimes()
+	c.docker.EXPECT().CgroupVersion(c.ctx).Return(1, nil).AnyTimes()
+}
+
+func (c *createClusterValidationTest) expectEmptyFlux() {
+	c.flux = flux.NewFluxFromGitOpsFluxClient(nil, nil, nil, nil)
+}
+
+func (c *createClusterValidationTest) expectPreflightValidationsToPass() {
+	c.validator.EXPECT().PreflightValidations(c.ctx).Return(nil).AnyTimes()
+}
+
+func TestCreateClusterValidationsSuccess(t *testing.T) {
+	test := newValidateTest(t)
+	test.parseClusterSpecFromFile(t, "testdata/cluster_vsphere.yaml")
```

Comment:
do we need to parse from file or can we just generate the cluster.Spec directly in go code?

Reply:
Updated this to be generated within go code and removed config file